realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -761,6 +793,19 @@ public void setList(String fieldName, RealmList<DynamicRealmObject> list) {
     }
 
     /**
+     * Sets the reference to a {@link RealmList} on the given field.
+     *
+     * @param fieldName field name.
+     * @param list list of references.
+     * @throws IllegalArgumentException if field name doesn't exist, it is not a list field, the type
+     * of the object represented by the DynamicRealmObject doesn't match or any element in the list belongs to a
+     * different Realm.
+     */
+    public void setValueList(String fieldName, RealmList<?> list) {
```

Comment:
Yes.
I added this because I didn't find a way to avoid `getValueList()`.

If we can omit `getValueList()`, I'll remove this.